server/lib/konnector_poller.coffee

```diff
@@ -18,187 +18,154 @@ month = 30 * day
 format = "DD/MM/YYYY [at] HH:mm:ss"
 periods = {hour: hour, day: day, week: week, month: month}
 timeouts = {}
+nextUpdates = {}
+timeout = null
 
 class KonnectorPoller
 
-    start: ->
+    start: (reset=false, cb=null) ->
         log.debug "Launching Konnector Poller..."
-        @init()
 
-    init: ->
-        Konnector.all (err, konnectors) =>
-
-            async.eachSeries konnectors, (konnector, callback) =>
-
-                # if both importInterval and lastAutoImport are valid
-                if konnector.importInterval? \
-                and konnector.importInterval isnt 'none' \
-                and konnector.lastAutoImport? \
-                and fs.existsSync path.resolve("server/konnectors/#{konnector.slug}.coffee")
-
-                    importInterval = periods[konnector.importInterval]
-                    now = moment()
-                    lastImport = moment(konnector.lastImport)
-                    lastAutoImport = moment(konnector.lastAutoImport)
-
-                    # if we missed an importation cycle
-                    if (now.valueOf() - lastAutoImport.valueOf()) > importInterval
-                        log.debug "#{konnector.slug} missed an importation cycle"
-
-                        # calculate the supposed last Auto-import
-                        importTime = lastAutoImport.valueOf() + importInterval
-
-                        # calculate the time elapsed
-                        while importTime < now.valueOf()
-                            importTime += importInterval
-                        interval = importTime - now.valueOf()
-
-                        # We import now
-                        importer konnector
-
-                        nextUpdate = now.clone()
-                        nextUpdate = nextUpdate.add interval, 'ms'
-                        log.debug "#{konnector.slug} | Next update : " +
-                        "#{nextUpdate.format(format)}"
-
-                        @prepareNextCheck konnector, interval
+        if reset
+            nextUpdates = {}
+        if Object.keys(nextUpdates).length is 0
+            Konnector.all (err, konnectors) =>
+                async.eachSeries konnectors, (konnector, callback) =>
+                    # if both importInterval and lastAutoImport are valid
+                    if konnector.importInterval? \
+                    and konnector.importInterval isnt 'none' \
+                    and konnector.lastAutoImport? \
+                    and fs.existsSync path.resolve("server/konnectors/#{konnector.slug}.coffee")
+                        nextUpdate = @findNextUpdate(konnector)
+                        nextUpdates[konnector.slug] = [nextUpdate, konnector]
                         callback()
```

Comment:
I suggest you extract the iterator to a class method like `initializeKonnectorPolling` or something (I don't have a good name)